### PR TITLE
Automate release steps post changelog PR merge.

### PR DESCRIPTION
**Pull request by gsquared94**
Fixes: #4324  

**Description**

Current release has a lot of manual actions that can be broadly categorized into two steps:
- Create Release Changelog PR
- Create and Publish Github Release

This PR automates the second step. When the Changelog PR is merged it automatically creates the Github Release(as a draft) with all text and attachments. We will only need to hit publish button once we're ok with the draft. In time this manual verification can be skipped too.

**User facing changes (remove if N/A)**

This requires that the Changelog PR is labelled with `draft-release` *before* being merged. Also the PR title should be `Release vX.x.x` or `vX.x.x release` since it parses the PR title for the tag semVer. 
**Follow-up Work (remove if N/A)**

In a subsequent PR I'll automate the first step of creating the Release Changelog PR.

### CODE REVIEW

File: .github/workflows/draft-release.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        echo ::set-env name=TAG_NAME::"v${t}"
+        echo ::set-env name=RELEASE_NAME::"v${t} Release"
+
+    - name: Install gsutil
+      run: |
+        curl -Lo $HOME/gsutil.tar.gz https://storage.googleapis.com/pub/gsutil.tar.gz
+        tar xfz $HOME/gsutil.tar.gz -C $HOME
+        echo ::add-path::$HOME/gsutil
+
+    - name: Download release artifacts
```
Comment (tejal29) [inline .github/workflows/draft-release.yml:33]: What happens when this step fails tp download anything or one of the binaries is missing?

File: .github/workflows/draft-release.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    - name: Create Release
+      shell: bash
+      run: |
+        curl -fsSL https://github.com/github/hub/raw/master/script/get | bash -s 2.14.1
+        body=$(git log -p --follow -1 CHANGELOG.md | grep '^\+' | cut -c 2- | tail -n +2)
+        assets=()
+        for asset in $HOME/${{ env.TAG_NAME }}/*; do
+        assets+=("-a" "$asset")
+        done
```
Comment (tejal29) [inline .github/workflows/draft-release.yml:54]: shd we assert here the number of binaries?

File: .github/workflows/draft-release.yml (hunk 1 above)
Comment (gsquared94) [reply .github/workflows/draft-release.yml:33]: The pipeline overall should fail as a result

File: .github/workflows/draft-release.yml (hunk 2 above)
Comment (gsquared94) [reply .github/workflows/draft-release.yml:54]: The number of artifacts is unfortunately already hardcoded on line 38: `$found -lt 10`

File: .github/workflows/draft-release.yml (hunk 1 above)
Comment (gsquared94) [reply .github/workflows/draft-release.yml:33]: image

File: .github/workflows/draft-release.yml (hunk 2 above)
Comment (gsquared94) [reply .github/workflows/draft-release.yml:54]: The `hub` command does log the number of artifacts that it uploads
image

**Comment by gsquared94** [comment]
> To test this, shd i create a PR with the label mentioned?

You should be able to test this by resetting your fork master with this branch and raising a PR that has changes to the `CHANGELOG.md` file; title like `Release v1.11.0` and the label `draft-release`. It'll only work if you try on an already released version since running it in your fork won't trigger the cloud build pipeline that creates the artifacts. You should delete the tag and release from the Github release page to verify that it recreates both. Or if you want to you can make it point to your own bucket and release whatever version you want.

**Comment by tejal29** [comment]
Thanks, i was able to test of the workflow got triggered 
image

**Comment by nkubala** [comment]
> This requires that the Changelog PR is labelled with draft-release before being merged

what happens if we forget to do this? inevitably I think it will happen at some point so we should know what our recourse is.

**Comment by tejal29** [comment]
This is great!
image

**Comment by tejal29** [comment]
and it created a draft release here 
image

**Comment by gsquared94** [comment]
> 
> 
> > This requires that the Changelog PR is labelled with draft-release before being merged
> 
> what happens if we forget to do this? inevitably I think it will happen at some point so we should know what our recourse is.

@nkubala ya I didn't want it to be automatically figured like just detecting changes to the Changelog file or a specific PR title as those could result in unwanted deployments. So it's explicit using a label. Unfortunately once the PR has been merged without the label, adding it to the closed PR won't help and Github Actions doesn't support manually triggering the workflow. The resolution would be another PR with an empty commit (sigh).

**Pull request merged**